src/main/java/com/minecolonies/commands/SingleCommand.java

```diff
@@ -0,0 +1,32 @@
+package com.minecolonies.commands;
+
+import net.minecraft.command.ICommandSender;
+import org.jetbrains.annotations.NotNull;
+
+
+/**
+ * A command that has children.
+ */
+public abstract class SingleCommand implements ISubCommand
+{
+
+    private final String[] parents;
+
+    public SingleCommand(@NotNull String[] parents)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AUseVarargs)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)